### PR TITLE
Fix`Ior` deprecation replacement expressions

### PR DESCRIPTION
**Pull request by gutiory**
This PR fixes some of the deprecation `ReplaceWith` suggestions for `Ior` listed in https://github.com/arrow-kt/arrow/issues/2964. I'll update the issue table once this is merged.

And for those replacements that are no working, a section will be created in Arrow website, explaining the best way to migrate these methods.

### CODE REVIEW

**Comment by rafaparadela** [comment]
Task linked: [CU-865byugz2 Fix Ior deprecation replacement expressions](https://app.clickup.com/t/865byugz2)

**Review by serras — APPROVED** [review]

File: arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Ior.kt [hunk 1]
```diff
@@ -1326,21 +1187,25 @@ public fun <A, C, B : C> Ior<A, B>.widen(): Ior<A, C> =
...
+    "arrow.typeclasses.combine"
+  )
 )
 public fun <A, B, C> Ior<A, B>.zip(SA: Semigroup<A>, fb: Ior<A, C>): Ior<A, Pair<B, C>> =
   ior(SA::combine) { Pair(this@zip.bind(), fb.bind()) }
 
 @Deprecated(
   NicheAPI + "Prefer using the inline ior DSL",
-  ReplaceWith("ior(SA::combine) { map(this@zip.bind(), c.bind()) }", "arrow.core.raise.ior", "arrow.typeclasses.combine")
+  ReplaceWith("ior({a, b -> a + b}) { map(this.bind(), c.bind()) }", "arrow.core.raise.ior")
```
Comment (franciscodr) [inline arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Ior.kt:1208]: I think @serras has asked this in another PR, but being `A` a generic type, this code won't compile unless we run it under the scope of a `Semigroup` instance, will it? 

Additionally, why do you use the `SA::combine` function in this `ReplaceWith` [expression](https://github.com/arrow-kt/arrow/pull/2976/files#diff-c5990e586847dc3242060efe9f70a256299b84edf33bdb0171486a3db9c5184bR1198) and use the `plus` operator in the rest of `zip` alternatives?

File: arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Ior.kt [hunk 2]
```diff
@@ -263,10 +263,7 @@ public sealed class Ior<out A, out B> {
     return fold({ f(c, it) }, { g(c, it) }, { a, b -> g(f(c, a), b) })
   }
 
-  @Deprecated(
-    NicheAPI + "Prefer when or fold instead",
-    ReplaceWith("  MN.run { fold({ f(it) },{ g(it) },{ a,b -> f(a).combine(g(b)) })}")
-  )
+  @Deprecated(NicheAPI + "Prefer when or fold instead. See the Arrow web migration guide for more info.")
```
Comment (franciscodr) [inline arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Ior.kt:266]: Have we created a task for adding these changes to the migration guide?

**Review by franciscodr — APPROVED** [review]
Just a couple of questions. Looks good to me

File: arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Ior.kt (hunk 2 above)
Comment (gutiory) [reply arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Ior.kt:266]: I've just created it 😄 . Thanks for the suggestion, @franciscodr!

File: arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Ior.kt (hunk 1 above)
Comment (gutiory) [reply arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Ior.kt:1208]: This was actually a bug 😅 . Good catch, @franciscodr !. See [`133847c` (#2976)](https://github.com/arrow-kt/arrow/pull/2976/commits/133847c46a24d3c2b6482009d962b0d4917fb62b)

**Pull request merged**